### ISSUE TITLE
What are requirements for using the max_rss setting?

### ISSUE THREAD

**Issue by philgyford** [GDay/django-q2#45]
The [docs for the `max_rss` config setting](https://django-q2.readthedocs.io/en/master/configure.html#max-rss) say:

> The maximum resident set size in kilobytes before a worker will recycle and release resources. Useful for limiting memory usage. Only supported on platforms that implement the python resource module or install the [psutil](https://django-q2.readthedocs.io/en/master/install.html#psutil-package) module. Defaults to `None`.

Forgive my ignorance but what is "the python resource module" and how do I know whether my platform (OS? Python version?) supports it?

Assuming I can use it, any suggestion for a reasonable limit to set? I have no idea what's "normal".

**Comment by GDay** [comment]
> Forgive my ignorance but what is "the python resource module" and how do I know whether my platform (OS? Python version?) supports it?

The Python resource module is this: https://docs.python.org/3/library/resource.html

> [Availability](https://docs.python.org/3/library/intro.html#availability): not Emscripten, not WASI.
> This module does not work or is not available on WebAssembly platforms wasm32-emscripten and wasm32-wasi. See [WebAssembly platforms](https://docs.python.org/3/library/intro.html#wasm-availability) for more information.

I think most people will have it available.

> Assuming I can use it, any suggestion for a reasonable limit to set? I have no idea what's "normal".

That's a little hard to answer. This entirely depends on your set up. 
The `max_rss` setting is really useful when you are having some kind of memory leak. If you get above a certain amount of memory, then release the worker and recreate one. 

This StackExchange answer probably gives a little more context: https://unix.stackexchange.com/a/30941

So basically: take the amount of memory you have in your environment. Then take some off for the main django q2 processes (like monitoring, scheduling tasks etc) and take some more off for the base OS and other processes running. Then what's left, divide over the amount of workers that you are running and that should be it.

Hope that helps.

**Comment by philgyford** [comment]
Great, thank you for the helpful answer! I'll keep my eye on what it tends to use in case I need to limit it.
